src/github.com/mozilla-services/pushgo/simplepush/emcee_store.go

```diff
@@ -544,7 +547,10 @@ func (s *EmceeStore) FetchAll(uaid string, since time.Time) ([]Update, []string,
 	keys := make([]string, 0, 20)
 
 	for _, chid := range chids {
-		key, _ := s.IDsToKey(uaid, chid)
+		key, err := s.IDsToKey(uaid, chid)
+		if err != nil {
```

Comment:
IDsToKey is/was a wrapper to store the UAID, CHID combination as a unique ID. Having this be determined by the storage system makes sense, because the ID may well be the primary key for that storage system. An error might occur if the "key" could not be rendered using the values (e.g. UAID or CHID is blank or contains values that are invalid like :doughnut: ).  

For what it's worth, if an error occurs here, I'd probably kick the transaction as invalid.